FreeRTOS_RA.c

```diff
@@ -89,8 +89,9 @@
              pxEndPoint != NULL;
              pxEndPoint = FreeRTOS_NextEndPoint( pxInterface, pxEndPoint ) )
         {
+            /* Check if it has the link-local prefix FE80::/10 */
             if( ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 0 ] == 0xfeU ) &&
-                ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 1 ] == 0x80U ) )
+                ( ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 1 ] & 0xc0U ) == 0x80U ) )
```

Comment:
Although almost all IPv6 link-local IP-addresses start with 0xfe80, the lower 6 bits of the second byte may be non-zero. Therefore I added an "and" and some comment.